### PR TITLE
Add sent and received bytes to listNetworks and listVirtualMachines.

### PR DESCRIPTION
**Pull request by ravening**
### Description

Display the traffic data in networks and vm api response so that we can quickly get the traffic statistics about all the vm's in that network. User can traffic data for a particular network through api as well.
Also users can monitor the traffic on individual Vm as well which was not possible earlier.

```
(local) mgt01 > list virtualmachines filter=receivedbytes,sentbytes,
{
  "count": 2,
  "virtualmachine": [
    {
      "receivedbytes": 187686,
      "sentbytes": 99809
    },
    {
      "receivedbytes": 189086,
      "sentbytes": 90855
    }
  ]
}
```

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [X] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [X] Trivial

### CODE REVIEW

**Review by DaanHoogland — APPROVED** [review]
clgtm needs testing, can you add info about the use-case to the description or refer an issue (even though it is trivial, it might hekp tracking if someone asks about this)

**Comment by ravening** [comment]
> clgtm needs testing, can you add info about the use-case to the description or refer an issue (even though it is trivial, it might hekp tracking if someone asks about this)

@DaanHoogland added more details in description

File: server/src/main/java/com/cloud/api/ApiResponseHelper.java [hunk 1]
```diff
@@ -394,6 +397,8 @@
     private VMSnapshotDao vmSnapshotDao;
     @Inject
     private BackupOfferingDao backupOfferingDao;
+    @Inject
+    private UserStatisticsDao _userStatsDao;
```
Comment (GabrielBrascher) [inline server/src/main/java/com/cloud/api/ApiResponseHelper.java:401]: @ravening can you please remove the underscore `_` at the beginning of the variable name?
Many variables on the codebase have `_` at the beginning but it is not aligned with the [Java variable naming convention](https://docs.oracle.com/javase/tutorial/java/nutsandbolts/variables.html).

According to [Java variable naming convention](https://docs.oracle.com/javase/tutorial/java/nutsandbolts/variables.html):

> A similar convention exists for the underscore character; while it's technically legal to begin your variable's name with "_", this practice is discouraged. White space is not permitted.

File: server/src/main/java/com/cloud/api/query/dao/UserVmJoinDaoImpl.java [hunk 2]
```diff
@@ -78,6 +81,8 @@
     private UserDao _userDao;
     @Inject
     private NicExtraDhcpOptionDao _nicExtraDhcpOptionDao;
+    @Inject
+    UserStatisticsDao _userStatsDao;
```
Comment (GabrielBrascher) [inline server/src/main/java/com/cloud/api/query/dao/UserVmJoinDaoImpl.java:85]: I don't want to sound repetitive and I would not block a PR due to `_`, in any circumstance. But I would really appreciate it if you could remove `_` from any added variable :slightly_smiling_face:

**Review by GabrielBrascher — COMMENTED** [review]
Thanks for the PR @ravening. Overall it looks good (code-wise at least, I did not test it).

I raised a couple of points. Can you please take a look at them @ravening?

**Comment by davidjumani** [comment]
@ravening Can you fix the conflicts? Thanks

**Comment by nvazquez** [comment]
Hi @ravening there is still a conflict with this PR can you please resolve it?

File: server/src/main/java/com/cloud/api/ApiResponseHelper.java (hunk 1 above)
Comment (ravening) [reply server/src/main/java/com/cloud/api/ApiResponseHelper.java]: @GabrielBrascher changed the code. please review again

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Review by nvazquez — APPROVED** [review]
LGTM

Tested on VM on shared network:

````
(localcloud) SBCM5> > list networks filter=id,receivedbytes,sentbytes,
{
  "count": 1,
  "network": [
    {
      "id": "8d4184b3-6e76-4fc6-bbdb-16e387a6efbe",
      "receivedbytes": 13519,
      "sentbytes": 9444
    }
  ]
}
(localcloud) SBCM5> > list virtualmachines filter=id,receivedbytes,sentbytes,
{
  "count": 1,
  "virtualmachine": [
    {
      "id": "1b954319-8a65-448b-bd1d-78f51ee2cd72",
      "receivedbytes": 13519,
      "sentbytes": 9444
    }
  ]
}
````

File: api/src/main/java/org/apache/cloudstack/api/response/UserVmResponse.java [hunk 3]
```diff
@@ -911,4 +919,12 @@ public String getOsDisplayName() {
     public String getPoolType() { return poolType; }
 
     public void setPoolType(String poolType) { this.poolType = poolType; }
+
+    public void setBytesReceived(final Long bytesReceived) {
```
Comment (RodrigoDLopez) [inline api/src/main/java/org/apache/cloudstack/api/response/UserVmResponse.java:923]: I don't fully understand why the key 'final' is being used here. You don't need to use it here.
am I wrong?

File: api/src/main/java/org/apache/cloudstack/api/response/UserVmResponse.java [hunk 4]
```diff
@@ -911,4 +919,12 @@ public String getOsDisplayName() {
...
+        this.bytesReceived = bytesReceived;
+    }
+
+    public void setBytesSent(final Long bytesSent) {
```
Comment (RodrigoDLopez) [inline api/src/main/java/org/apache/cloudstack/api/response/UserVmResponse.java:927]: and here.

File: server/src/main/java/com/cloud/api/query/dao/UserVmJoinDaoImpl.java [hunk 5]
```diff
@@ -365,9 +370,26 @@ public UserVmResponse newUserVmResponse(ResponseView view, String objectName, Us
             userVmResponse.setDynamicallyScalable(userVm.isDynamicallyScalable());
         }
 
+        addVmRxTxDataToResponse(userVm, userVmResponse);
+
         return userVmResponse;
     }
 
+    private void addVmRxTxDataToResponse(final UserVmJoinVO userVm, final UserVmResponse userVmResponse) {
```
Comment (RodrigoDLopez) [inline server/src/main/java/com/cloud/api/query/dao/UserVmJoinDaoImpl.java:378]: and again. here.

File: api/src/main/java/org/apache/cloudstack/api/response/NetworkResponse.java [hunk 6]
```diff
@@ -495,4 +503,12 @@ public Date getCreated() {
     public void setCreated(Date created) {
         this.created = created;
     }
+
+    public void setBytesReceived(final Long bytesReceived) {
```
Comment (RodrigoDLopez) [inline api/src/main/java/org/apache/cloudstack/api/response/NetworkResponse.java:507]: I don't fully understand why the key 'final' is being used here. You don't need to use it here.
am I wrong?

**Review by RodrigoDLopez — APPROVED** [review]
Code LGTM. 
But to me seems like you are using the key 'final' where we don't need it.

Did I understood something wrong?

Do we need more manual test on this PR? since Vazquez has already did some tests.

**Comment by DaanHoogland** [comment]
> Code LGTM.
> But to me seems like you are using the key 'final' where we don't need it.
> 
> Did I understood something wrong?
> 
> Do we need more manual test on this PR? since @nvazquez has already did some tests.

It is, in some circles, good practice to make sure a parameter is only changed consciously. A bit overdone indeed, i would agree.

**Comment by nvazquez** [comment]
Hi @ravening can you please address the open comments?

**Comment by ravening** [comment]
> Hi @ravening can you please address the open comments?

@nvazquez done

**Comment by nvazquez** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

**Pull request merged**